File: ThreadingSimulator/Models/CommandModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using ThreadingSimulator.Enums;
using ThreadingSimulator.Other;

namespace ThreadingSimulator.Models
{
    public class CommandModel : PropertyChangedImpl
    {
        public CommandType Type { get; set; }
        public string Variable { get; set; }
        public int? Value { get; set; }

        #region Display
        private bool displayColor;

        public bool DisplayColor
        {
            get
            {
                return displayColor;
            }
            set
            {
                displayColor = value;
                OnPropertyChanged();
            }
        }
        #endregion

        public CommandModel Clone()
        {
            return MemberwiseClone() as CommandModel;
        }
    }
}

File: GUI/src/app/Models/CommandModel.ts
import { CommandType } from "../Enums/CommandType";

export class CommandModel
{
    public Type: CommandType;
    public Variable: string;
    public Value: number;

    public DisplayColor: boolean;

    
    public Clone(): CommandModel
    {
        return CommandModel.Deserialize(this);
    }

    public static Deserialize(obj: any): CommandModel
    {
        if(obj == null)
        {
            return null;
        }

        let trueObj: CommandModel = new CommandModel();

        trueObj.DisplayColor = obj.DisplayColor;
        trueObj.Type = obj.Type;
        trueObj.Value = obj.Value;
        trueObj.Variable = obj.Variable;

        return trueObj;
    }
}